### PR TITLE
[17.0][FW] sale_discount_display_amount: multiple ports from 14.0

### PR DESCRIPTION
**Pull request by henrybackman**
Port of the following PRs from 14.0 to 17.0:
- #2071
- #2592
- #2673
- #2726

### CODE REVIEW

File: sale_discount_display_amount/__manifest__.py [hunk 1]
```diff
@@ -11,7 +11,10 @@
     "author": "ACSONE SA/NV,Odoo Community Association (OCA)",
     "website": "https://github.com/OCA/sale-workflow",
     "depends": ["sale_management"],
-    "data": ["views/sale_view.xml", "views/res_config_settings_views.xml"],
+    "data": [
+        "views/sale_view.xml",
+        "report/sale_report_template.xml",
+    ],
```
Comment (mmequignon) [inline sale_discount_display_amount/__manifest__.py]: we are missing the config view here

File: sale_discount_display_amount/migrations/14.0.1.2.0/post-compute-subtotal-columns.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import logging
```
Comment (mmequignon) [inline sale_discount_display_amount/migrations/14.0.1.2.0/post-compute-subtotal-columns.py]: I don't think we need those 14.0.1.2.0 migration scripts

File: sale_discount_display_amount/models/sale_order_line.py [hunk 3]
```diff
@@ -40,6 +40,7 @@ def _update_discount_display_fields(self):
             if not line.discount:
                 line.price_subtotal_no_discount = line.price_subtotal
                 line.price_total_no_discount = line.price_total
+                line.price_subtotal_no_discount = line.price_subtotal
```
Comment (mmequignon) [inline sale_discount_display_amount/models/sale_order_line.py]: This looks already there at line 41, do we need this commit to be ported at all?

**Review by mmequignon — CHANGES_REQUESTED** [review]

File: sale_discount_display_amount/migrations/14.0.1.2.0/post-compute-subtotal-columns.py (hunk 2 above)
Comment (simahawk) [reply sale_discount_display_amount/migrations/14.0.1.2.0/post-compute-subtotal-columns.py]: We should not "remove" them, we should change the version to `17.0.1.1.0`.
Just do a `git move` and a separated commit like `$mod_name: adapt migration scripts to v17`

**Review by simahawk — CHANGES_REQUESTED** [review]

File: sale_discount_display_amount/__manifest__.py (hunk 1 above)
Comment (henrybackman) [reply sale_discount_display_amount/__manifest__.py]: Added

File: sale_discount_display_amount/migrations/14.0.1.2.0/post-compute-subtotal-columns.py (hunk 2 above)
Comment (henrybackman) [reply sale_discount_display_amount/migrations/14.0.1.2.0/post-compute-subtotal-columns.py]: Migration scripts moved to `17.0.1.1.0` in a separate commit

File: sale_discount_display_amount/models/sale_order_line.py (hunk 3 above)
Comment (henrybackman) [reply sale_discount_display_amount/models/sale_order_line.py]: Removed

**Review by simahawk — APPROVED** [review]

**Review by mmequignon — APPROVED** [review]

**Pull request merged**